tests/tests_fabric/strategies/test_xla_fsdp.py

```diff
@@ -0,0 +1,286 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from copy import deepcopy
+from unittest import mock
+from unittest.mock import MagicMock, Mock
+
+import pytest
+import torch
+import torch.nn as nn
+from torch.optim import Adam
+from torch.utils.data import DataLoader
+
+from lightning.fabric import Fabric
+from lightning.fabric.accelerators import XLAAccelerator
+from lightning.fabric.accelerators.xla import _XLA_AVAILABLE
+from lightning.fabric.strategies import XLAFSDPStrategy
+from lightning.fabric.strategies.xla_fsdp import _XLAFSDPBackwardSyncControl
+from tests_fabric.helpers.models import RandomDataset
+from tests_fabric.helpers.runif import RunIf
+
+if _XLA_AVAILABLE:
+    import torch_xla.core.xla_model as xm
+    from torch_xla.distributed.fsdp.wrap import always_wrap_policy
+    from torch_xla.distributed.fsdp.xla_fully_sharded_data_parallel import XlaFullyShardedDataParallel
+else:
+    xm = None
+    always_wrap_policy = None
+    XlaFullyShardedDataParallel = None
```

Comment:
Can you make these all local imports in their respective tests? If we support XLA with GPU, xla would be available and then they would get imported even if there's no TPU in the environment